### PR TITLE
feat: implement histogram type view for policy violation types on all…

### PR DESCRIPTION
**Pull request by maurycupitt**
… versions.  This is meant to offer a view into the different types of policy violations each version violates.

image

### CODE REVIEW

File: src/components/Popup/IQServer/RemediationPage/AllVersionsPage/AllVersionsDetails/AllVersionsDetails.tsx [hunk 1]
```diff
@@ -27,26 +27,27 @@ import { ExtensionPopupContext } from '../../../../../../context/ExtensionPopupC
...
-        if (policyData.policyViolations && policyData.policyViolations.length > 0) {
-            return Math.max(
-                ...policyData.policyViolations.map((violation) =>
-                    violation.threatLevel != undefined ? violation.threatLevel : 0
-                )
-            )
-        }
-        return 0
-    }
+    // function getMaxViolation(policyData: ApiComponentPolicyViolationListDTOV2) {
```
Comment (madpah) [inline src/components/Popup/IQServer/RemediationPage/AllVersionsPage/AllVersionsDetails/AllVersionsDetails.tsx:41]: Commented code should be removed @maurycupitt

File: src/components/Popup/IQServer/RemediationPage/AllVersionsPage/AllVersionsDetails/AllVersionsDetails.tsx [hunk 2]
```diff
@@ -58,14 +59,23 @@ function IqAllVersionDetails() {
...
+    const formatDate = (date: Date | undefined | null): string => {
+        if (date) {
+            const dateTime = new Date(date)
+            const noTime = dateTime.toUTCString().split(' ').slice(0, 4).join(' ')
+            return noTime
+        }
+        return 'N/A'
+    }
+
+    // const VersionTooltip = withStyles((theme) => ({
```
Comment (madpah) [inline src/components/Popup/IQServer/RemediationPage/AllVersionsPage/AllVersionsDetails/AllVersionsDetails.tsx:71]: Remove commented code.

**Review by madpah — CHANGES_REQUESTED** [review]
Changes as per comments @maurycupitt

File: docs/images/extension-open-01.png
Comment (madpah) [inline docs/images/extension-open-01.png:1]: Have we purposefully lost information on the Component Page here @maurycupitt ?

- First Release Date
- Last Release Date

**Review by madpah — COMMENTED** [review]
See comments against updated screenshots.

File: docs/images/extension-open-01.png
Comment (maurycupitt) [reply docs/images/extension-open-01.png:1]: Confirmed that we removed them on purpose

File: docs/images/extension-open-02.png
[the remark this answers is not in the document]
Comment (maurycupitt) [reply docs/images/extension-open-02.png:1]: It is going to be less compact as I added the highest levels for each of the policy categories

**Pull request merged**